packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts

```diff
@@ -0,0 +1,95 @@
+import { gql } from "apollo-server-express";
+
+export const knowledgeEntityTypedef = gql`
+  interface KnowledgeEntity {
+    # These fields are repeated everywhere they're used because
+    # (a) GQL requires it - https://github.com/graphql/graphql-spec/issues/533
+    # (b) string interpolation breaks the code generator's introspection
+    #
+    # Could maybe use a custom schema loader to parse it ourselves:
+    # https://www.graphql-code-generator.com/docs/getting-started/schema-field#custom-schema-loader
+    #
+    # For now, _COPY ANY CHANGES_ from here to any type that 'implements Entity'
+    """
+    The id of the entity
+    """
+    entityId: ID!
+    """
+    The specific version of the entity
+    """
+    entityVersionId: String!
+    """
+    The id of the account that owns this entity.
+    """
+    ownedById: ID!
+    """
+    The fixed id of the type this entity is of.
+    """
+    entityTypeVersionedUri: ID!
+    """
+    The full entity type definition.
+    """
+    entityType: EntityType
+    """
+    The linked entities of the entity.
+    """
+    linkedEntities: [KnowledgeEntity!]!
+    """
+    The JSON object containing the entity's properties.
+    """
+    properties: JSONObject!
+  }
+
+  """
+  For referring to an existing entity owned by a specific account id
+  """
+  input KnowledgeExistingEntity {
+    """
+    This may be a reference to a placeholder set using placeholderId on a previous UpdatePageContentsAction.
+    """
+    entityId: ID!
+    ownedById: ID!
+  }
+
+  """
+  Select entity types by ONE of componentId, entityTypeVersionedUri
+  """
+  input KnowledgeEntityTypeChoice {
+    """
+    For entity types related to block types, the URI of the block. 'componentId' in the entity type's schema.
+    """
+    componentId: ID
+    """
+    A fixed entity type ID. This may be a reference to a placeholder set using a previous createEntityTypeAction.
+    """
+    entityTypeVersionedUri: String
+  }
+
+  input KnowledgeLinkedEntityDefinition {
+    destinationAccountId: ID!
+    """
+    The index of the link (if any)
+    """
+    index: Int
+    entity: KnowledgeEntityDefinition!
+  }
+
+  input KnowledgeEntityDefinition {
+    """
+    Existing Entity to use instead of creating a new entity.
+    """
+    existingEntity: KnowledgeExistingEntity
+    """
+    The type of which to instantiate the new entity.
+    """
+    entityType: KnowledgeEntityTypeChoice
```

Comment:
**Blocking:** why is this optional? Shouldn't an entity type always be specified when creating an entity?
```suggestion
    """
    The type of which to instantiate the new entity.
    """
    entityType: KnowledgeEntityTypeChoice!
```

Reply:
This is sort of a result of the way this was in the `main` branch API, where an `EntityDefinition` can be a reference to an existing entity, in which case the entity type doesn't need to be given

Reply:
Ah I missed that, thanks for explaining 👍